rpi_backlight.py

```diff
@@ -38,97 +39,174 @@ def _get_value(name: str) -> str:
+    elif mode == "MODE_RPI":
+        if not -10 < value <= max_value:
+                raise ValueError(
+                    "value must be between 0 and {}, got {}".format(max_value, value)
+                )
 
     if smooth:
         if not isinstance(duration, (int, float)):
             raise ValueError(
                 "integer or float required, got '{}'".format(type(duration))
             )
-        actual = get_actual_brightness()
+        actual = get_brightness_value()
         diff = abs(value - actual)
         while actual != value:
             actual = actual - 1 if actual > value else actual + 1
-
-            _set_value("brightness", actual)
+            set_brightness_value(actual)
             time.sleep(duration / diff)
     else:
-        _set_value("brightness", value)
+        set_brightness_value(value)
 
 
-def set_power(on: bool) -> None:
+def set_power(on: bool, smooth: bool = True, duration: float = 1) -> None:
     """Set the display power on or off.
-
     :param on: Boolean whether the display should be powered on or not
     """
-
     # 0 is on, 1 is off
-    _set_value("bl_power", int(not on))
-
+    if mode == "MODE_RPI":
+        _set_value("bl_power", int(not on))
+    else:
+        if on:
+            value = 255
+        else: 
+            value = 0
+        if smooth:
+            if not isinstance(duration, (int, float)):
+                raise ValueError(
+                    "integer or float required, got '{}'".format(type(duration))
+                )
+            actual = get_brightness_value()
+            diff = abs(value - actual)
+            while actual != value:
+                actual = actual - 1 if actual > value else actual + 1
+                set_brightness_value(actual)
+                time.sleep(duration / diff)
+        else:
+            set_brightness_value(value)
+
+
+def toggle_power(smooth: bool = True, duration: float = 1) -> None:
+    """Toggle the display power on or off."""
+    if mode == "MODE_RPI":
+        set_power(not get_power())
+    elif mode == "MODE_TINKERBOARD":
+        if int(_get_value("tinker_mcu_bl")) == 0:
+            value = 255
+        else:
+            value = 0
+    if smooth:
+        if not isinstance(duration, (int, float)):
+            raise ValueError(
+                "integer or float required, got '{}'".format(type(duration))
+            )
+        actual = get_brightness_value()
+        diff = abs(value - actual)
+        while actual != value:
+            actual = actual - 1 if actual > value else actual + 1
+            set_brightness_value(actual)
+            time.sleep(duration / diff)
+    else:
+        set_brightness_value(value)
 
 def _create_argument_parser() -> argparse.ArgumentParser:
     parser = argparse.ArgumentParser(
         description="Control power and brightness of the "
         'official Raspberry Pi 7" touch display.'
     )
-    parser.add_argument(
+    if mode == "MODE_TINKERBOARD":
```

Comment:
Again, since you've decided to change the min brightness value to 0 there's no need for this.

Reply:
fixed